src/main/java/mate/academy/springboot/datajpa/service/impl/ProductServiceImpl.java

```diff
@@ -36,7 +34,7 @@ public void delete(Long id) {
 
     @Override
     public Product update(Product product) {
-        return productRepository.saveAndFlush(product);
+        return productRepository.save(product);
```

Comment:
Check if a category exists before updating OR delete this method at all and use the method `save` from the service